src/lib/server/eliommod_sessadmin.ml

```diff
@@ -211,8 +209,9 @@ let update_pers_exp full_st_name sitedata old_glob_timeout new_glob_timeout =
           | Some t when t <= now ->
               Eliommod_persess.close_persistent_state2
                 ~scope sitedata sessgrp k
-          | _ -> Ocsipersist.add table k (full_st_name2, newexp,
-                   Eliom_common.TGlobal, sessgrp) >>= Lwt_unix.yield
+          | _ ->
+            Eliom_common.Persistent_cookies.Cookies.add k
```

Comment:
Don't you need to update `Eliom_common.Expiry_dates` as well?
(May be it should be done in `Eliom_common.Persistent_cookies.Cookies.add`?

Reply:
Done (more systematically than before).